src/coreclr/jit/scopeinfo.cpp

```diff
@@ -403,26 +403,12 @@ void CodeGenInterface::siVarLoc::siFillRegisterVarLoc(
             break;
 #endif // !TARGET_64BIT
 
-#ifdef TARGET_64BIT
         case TYP_FLOAT:
         case TYP_DOUBLE:
-            // TODO-AMD64-Bug: ndp\clr\src\inc\corinfo.h has a definition of RegNum that only goes up to R15,
-            // so no XMM registers can get debug information.
```

Comment:
This comment is still true (though RegNum is no longer is corinfo.h, apparently).

See `CodeGen::checkICodeDebugInfo()`. We set `vlrReg` and assume that our register number values are the same as the RegNum values. `RegNum` doesn't define the FP registers, but we still set it. Seems odd. I presume the debugger ignores it?

Reply:
Well the comment might be true, but I think the real problem is that the current `RegNumber` is actually `GeneralPurposeRegNumber`, and the value stored in `vlReg.vlrReg` for `VLT_REG_FP` is some `FloatingPointerRegNumber` enum that doesn't exist (i.e. xmm0 gets index 0). The debugger handles it here for x64:
https://github.com/dotnet/runtime/blob/efe8b8789b0ef472c21e6ea934d2dc4481719283/src/coreclr/debug/di/rsthread.cpp#L8410-L8411

I think I'll need to add some similar handling for x86 and run the debugger tests as part of this PR.

Reply:
This needs to be checked in the managed ObjWriter too since it expects the JIT register number (ie. not zero based).